common/djangoapps/mailchimp_pipeline/tests/test_tasks.py

```diff
@@ -0,0 +1,283 @@
+from lms.djangoapps.certificates import api as certificate_api
+from xmodule.modulestore.tests.factories import CourseFactory
+from student.tests.factories import UserProfileFactory
+from student.models import CourseEnrollment
+
+from mailchimp_pipeline.tasks import update_org_details_at_mailchimp, update_enrollments_completions_at_mailchimp
+from mailchimp_pipeline.helpers import get_user_active_enrollements, get_enrollements_course_short_ids
+from mailchimp_pipeline.client import MailChimpException
+from mailchimp_pipeline.signals.handlers import (
+    send_user_info_to_mailchimp,
+    sync_metric_update_prompt_with_mail_chimp,
+    update_mailchimp,
+    send_user_enrollments_to_mailchimp,
+    task_send_account_activation_email,
+    task_send_user_info_to_mailchimp,
+    send_user_course_completions_to_mailchimp
+)
+
+
+
+@override_settings(MAILCHIMP_LEARNERS_LIST_ID="test_id")
+class MailchimpPipelineTaskTestClass(ModuleStoreTestCase):
+    """
+        Tests for tasks and generic functions
+    """
+
+    ROOT_URL = "https://us12.api.mailchimp.com/3.0/"
+
+    @patch("nodebb.signals.handlers.get_current_request")
+    @patch("common.lib.mandrill_client.client.MandrillClient.send_mail")
+    def setUp(self, mocked_nodebb_request, mocked_email_sender):
+        super(MailchimpPipelineTaskTestClass, self).setUp()
+        patcher = patch('mailchimp_pipeline.client.request', autospec=True)
+        self.mock_request = patcher.start()
+        self.mock_request.return_value.status_code = 204
+        self.addCleanup(patcher.stop)
+
+        if User.objects.all().count() == 0:
+            self.user = UserFactory(is_staff=False, password='test')
+        else:
+            self.user.extended_profile = UserExtendedProfileFactory(user=self.user)
+            self.user.profile = UserProfileFactory.create(user=self.user)
+
+        organization = create_organization(self.user)
+        self.user.extended_profile.organization = organization
+        self.user.extended_profile.save()
+        course = CourseFactory.create(org="test", course="course", display_name="run")
+        course.save()
+        CourseEnrollment.enroll(self.user, course.id)
+
+    def test_update_mailchimp(self):
+        data = {'data': 'test_data'}
+        update_mailchimp(self.user.email, data)
+        expected_url = generate_mailchimp_url(self.ROOT_URL, self.user.email, settings.MAILCHIMP_LEARNERS_LIST_ID)
+        self.mock_request.assert_called_with(
+            "PUT", url=expected_url, headers=ANY, data=json.dumps(data), auth=ANY, params=ANY)
+
+    def test_update_mailchimp_for_exception(self):
+        self.mock_request.return_value.status_code = 404
+        self.mock_request.return_value.raise_for_status.side_effect = requests.exceptions.HTTPError
+        update_mailchimp(self.user.email, {})
+        self.assertRaises(MailChimpException)
+
+    @override_settings(CELERY_TASK_ALWAYS_EAGER=True, CELERY_TASK_EAGER_PROPOGATES=True)
```

Comment:
What do these settings do?

Reply:
To change Async tasks into Sync tasks so that I can test tasks result.

Reply:
Okay. What about overriding these settings on the class level? to avoid repetition.

Reply:
I guess you are right, that would be better. I will update this (y)